gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/zosmf/ZosmfService.java

```diff
@@ -220,9 +224,40 @@ public String getZosmfRealm(String infoURIEndpoint) {
         }
     }
 
+    @SuppressWarnings("java:S128") // Break in ZOWE case is intentionally written like this
+    public ZosmfResponse exchangeAuthenticationForZosmfToken(String token, AuthSource.Parsed authSource) {
+
+        String zosmfToken = token;
+        String cookieName = ZosmfService.TokenType.JWT.getCookieName();
+        switch (authSource.getOrigin()) {
+            case ZOSMF:
+                break;
+            case ZOWE:
+                zosmfToken = authenticationService.getLtpaToken(token);
+                if (zosmfToken != null) {
+                    cookieName = ZosmfService.TokenType.LTPA.getCookieName();
+                    break;
+                }
+            default:
+                Map<ZosmfService.TokenType, String> zosmfTokens = tokenCreationService.createZosmfTokensWithoutCredentials(authSource.getUserId());
+
+                if (zosmfTokens.containsKey(JWT)) {
+                    zosmfToken = zosmfTokens.get(JWT);
+                } else if (zosmfTokens.containsKey(LTPA)) {
+                    zosmfToken = zosmfTokens.get(LTPA);
+                    cookieName = ZosmfService.TokenType.LTPA.getCookieName();
+                } else {
+                    return null;
+                }
+        }
+
+        return new ZosmfResponse(cookieName, zosmfToken);
```

Comment:
Or.. maybe it could be more readable if the values were not collected but just returned everywhere we know them... like:

```java
    @SuppressWarnings("java:S128") // Break in ZOWE case is intentionally written like this
    public ZosmfResponse exchangeAuthenticationForZosmfToken(String token, AuthSource.Parsed authSource) {
        switch (authSource.getOrigin()) {
            case ZOSMF:
                return new ZosmfResponse(ZosmfService.TokenType.JWT.getCookieName(), token);
            case ZOWE:
                String ltpaToken = authenticationService.getLtpaToken(token);
                if (ltpaToken != null) {
                    return new ZosmfResponse(ZosmfService.TokenType.LTPA.getCookieName(), ltpaToken);
                }
            default:
                Map<ZosmfService.TokenType, String> zosmfTokens = tokenCreationService.createZosmfTokensWithoutCredentials(authSource.getUserId());

                if (zosmfTokens.containsKey(JWT)) {
                    return new ZosmfResponse(ZosmfService.TokenType.JWT.getCookieName(), zosmfTokens.get(JWT));
                } else if (zosmfTokens.containsKey(LTPA)) {
                    return new ZosmfResponse(ZosmfService.TokenType.LTPA.getCookieName(), zosmfTokens.get(LTPA));
                }
        }

        throw new ServiceNotFoundException("Cannot obtain z/OSMF token");
    }
```